pkg/azure/api/validation/validation.go

```diff
@@ -172,6 +190,16 @@ func validateOSDisk(osDisk api.AzureOSDisk, fldPath *field.Path) field.ErrorList
 	if osDisk.DiskSizeGB <= 0 {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("diskSizeGB"), osDisk.DiskSizeGB, "OSDisk size must be positive and greater than 0"))
 	}
+
+	if securityProfile := osDisk.ManagedDisk.SecurityProfile; securityProfile != nil {
```

Comment:
Can you add/update the unit tests for this piece of code?

Reply:
Addressed in https://github.com/gardener/machine-controller-manager-provider-azure/pull/146/commits/b0c6f389392c92a92c559ec5031d67bf06c206b7